# IMPORTANT: Parameters is_kf_optimization_enabled and SLAM_System is reset in the traversability_node.cpp when running through ROS and not directly with ORB-SLAM3.
# The updated parameters for this is taken from traversability_ros_params.yaml

# Choose from ORB3, ISAE
SLAM_System: "ORB3"

# --- Gridmap related ---
half_size_local_map: 80.5 # size of the entire local map. The individual maps will be stitched in this space.
grid_center_x: 0.0
grid_center_y: 0.0
resolution_local_map: 0.05
half_size_traversability: 78.5 # size of the map around the robot to consider for computing on the go traversability.
extend_length_every_resize_by: 30.5
use_least_squares_fit_to_compute_normals: true
use_pca_to_compute_normals: false

# --- Traversability related ---
security_distance: 0.40
ground_clearance: 0.15
max_slope: 0.8
robot_height: 5.5
translation_change_threshold: 0.25
rotation_change_threshold: 0.2
compute_roughness: true

# --- Mapping related ---
num_local_keyframes: 10
is_kf_optimization_enabled: true # if this is set to true, the pointclouds are stored in memory. Large pointclouds can lead to fast memory exhaustion.
use_pointcloud_buffer: true
use_ros_buffer: true
use_probabilistic_update: false
use_averaging: false
average_persistence: 10.0 #if a cell is averaged 20 times, it is reset to start a new averaging instance.
global_adjustment_sleep: 0 #time between processing every keyframe outside the local window (ms)
use_virtual_boundary: false